### DISCUSSION TITLE
How can a new user be created?

### DISCUSSION THREAD

**Discussion by slashtechno** [pocketbase/pocketbase#5293] [Q&A] [answered]
How can a user be created? I've looked through the [docs](https://pocketbase.io/docs/api-records/#create-record) but get one of two errors: 

When `userId` is not set (`{username: "abcxyz", password: "abcxyz", passwordConfirm: "abcxyz"}`):
```
{
    "code": 400,
    "message": "Something went wrong while processing your request.",
    "data": {
        "userId": {
            "code": "validation_required",
            "message": "Missing required value."
        }
    }
}
```

When `userId` is set to the username (`{userId: "abcxyz", username: "abcxyz", password: "abcxyz", passwordConfirm: "abcxyz"}`):
```
{
    "code": 400,
    "message": "Something went wrong while processing your request.",
    "data": {
        "userId": {
            "code": "validation_missing_users",
            "message": "Failed to fetch all users with the provided ids."
        }
    }
}
```

```js
    async function signUp() {
        try {
            const data = {
                // userId: username,
                username: username,
                password: password,
                passwordConfirm: password,
            };
            const createdUser = await pb.collection("profiles").create(data);
            await login();
        } catch (err) {
            console.error(err);
        }
    }
```

![image](https://github.com/user-attachments/assets/c13416ef-18ca-4d0b-a83c-a73948ecf92a)

**Comment by Irvingouj** [comment]
try 
```
pb.collection("users").create(data);
```
instead of 
```
await pb.collection("profiles").create(data);
```

**Reply by Irvingouj** [reply]
BTW, your table name starts with capital, bad practice, change it to all lowercase with under _

**Reply by Irvingouj** [reply]
If this did not help you, try look at your terminal, where you run your pocketbase, for more debugging informations

**Comment by Irvingouj** [comment]
Sorry! Let me correct my previous comments. I saw your problem this time.

**Comment by Irvingouj** [comment]
Make sure you created an auth collection

**Reply by slashtechno** [reply]
An auth collection? Isn't the `Users` section in the admin interface the built-in collection accessible under the collection `profiles`?

**Comment by Irvingouj** [comment]
![image](https://github.com/user-attachments/assets/dd4fb39c-77f7-4457-ac1a-c91e9c7d5652)

**Comment by ganigeorgiev** [comment, marked as answer]
@slashtechno From your screenshot you are using a very old PocketBase version (< 0.8). Please upgrade to the latest [v0.22.18](https://github.com/pocketbase/pocketbase/releases), the auth apis have changed significantly since v0.8.

**Reply by slashtechno** [reply]
I didn't realize that my Dockerfile specified an older version - I should have started debugging there. Thanks for pointing that out. I used the latest release and it works fine.
